cpp/src/arrow/ipc/read_write_test.cc

```diff
@@ -720,6 +771,11 @@ TEST_F(TestWriteRecordBatch, WriteWithCompression) {
     write_options.use_threads = false;
     read_options.use_threads = false;
     CheckRoundtrip(*batch, write_options, read_options);
+
+    ASSERT_OK_AND_ASSIGN(write_options.codec, MockCodec::Create(codec));
+    // 200% savings is impossible, so compression/decompression should be skipped
+    write_options.min_space_savings = 2.0;
```

Comment:
A more interesting check would be a `min_space_savings` between 0 and 1, but without random (uncompressible) data, to check that compression isn't enabled.

Reply:
(something like `min_space_savings = 2.0` should perhaps raise an error?)